src/walker/abstract_walker.js

```diff
@@ -351,13 +372,28 @@ sre.AbstractWalker.prototype.depth = function() {
 
 
 /**
+ * @return {string} The depth announcement for the currently focused element.
+ * @private
+ */
+sre.AbstractWalker.prototype.depth_ = function() {
+  var oldDepth = sre.Grammar.getInstance().getParameter('depth');
+  sre.Grammar.getInstance().setParameter('depth', true);
+  var snodes = this.focus_.getSemanticNodes();
+  var prefix = sre.SpeechGeneratorUtil.retrievePrefix(snodes[0]);
+  sre.Grammar.getInstance().setParameter('depth', oldDepth);
+  return this.levelAnnouncement_(prefix);
```

Comment:
Get rid of levelannouncement_. 
Have single depth_ method, analogous to summary/detail.
Use merge prefix method.